File: app/src/main/java/com/itba/runningMate/utils/providers/schedulers/AndroidTestSchedulerProvider.java
package com.itba.runningMate.utils.providers.schedulers;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

public class AndroidTestSchedulerProvider implements SchedulerProvider {

    @Override
    public Scheduler io() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler computation() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler ui() {
        return Schedulers.trampoline();
    }
}

File: app/src/main/java/com/itba/runningMate/utils/providers/schedulers/AndroidTestSchedulerProvider.kt
package com.itba.runningMate.utils.providers.schedulers

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

class AndroidTestSchedulerProvider : SchedulerProvider {

    override fun io(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun computation(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun ui(): Scheduler {
        return Schedulers.trampoline()
    }
}